wazp/utils.py

```diff
@@ -261,3 +266,107 @@ def export_selected_rows_as_yaml(
             yaml.dump(row, yamlf, sort_keys=False)
 
     return
+
+
+def get_dataframes_to_combine(
+    list_selected_videos: list,
+    slider_start_end_labels: list,
+    app_storage: dict,
+):
+    # build list of h5 files for the selected videos
+    # TODO: alternative to h5: pickle? csv?
+    # TODO: try to make it generic to any pose estim library?
+    # (right now DLC)
+    list_h5_file_paths = [
+        pl.Path(app_storage["config"]["pose_estimation_results_path"])
+        / pl.Path(
+            pl.Path(vd).stem
+            + app_storage["config"]["pose_estimation_model_str"]
+            + ".h5"
+        )
+        for vd in list_selected_videos
+    ]
+
+    # loop thru videos and h5 files to read dataframe and
+    # extract subset of rows
+    list_df_to_export = []
+    for h5, video in zip(list_h5_file_paths, list_selected_videos):
+
+        # get the metadata file for this video
+        # (built from video filename)
+        yaml_filename = pl.Path(
+            app_storage["config"]["videos_dir_path"]
+        ) / pl.Path(pl.Path(video).stem + ".metadata.yaml")
+
+        # extract the frame numbers
+        # from the slider position
+        with open(yaml_filename, "r") as yf:
+            metadata = yaml.safe_load(yf)
+            # extract frame start/end
+            frame_start_end = [
+                metadata["Events"][x] for x in slider_start_end_labels
+            ]
+            # extract ROI paths
+            # TODO
+            # ...
+
+        # read h5 as dataframe and add 'File' row
+        df = pd.concat(
+            [pd.read_hdf(h5)],
+            keys=[video],
+            names=[app_storage["config"]["metadata_key_field_str"]],
+            axis=1,
+        )
+
+        # add ROI and event tag per bodypart
+        # TODO: is there a nicer way using pandas?
+        # right now levels are hardcoded....
+        # TODO: ideally event_tags are not repeated per bodypart
+        # alternative; set as index? add frame from index?
+        # # df = df.set_index([df.index, "event_tag"])
+        metadata["Events"] = dict(
+            sorted(
+                metadata["Events"].items(),
+                key=lambda item: item[1],
+            )
+        )  # ensure keys are sorted by value
+        list_keys = list(metadata["Events"].keys())
+        list_next_keys = list_keys[1:]
+        list_next_keys.append("NAN")
+        for vd in df.columns.get_level_values(0).unique().to_list():
+            for scorer in df.columns.get_level_values(1).unique().to_list():
+                for bdprt in df.columns.get_level_values(2).unique().to_list():
```

Comment:
This is a very minor thing, but the current multi-level columns names are `File`, `scorer`, `bodyparts`, and `coords`. The `coords` is inhereted from `deeplabcut`, but is no longer appropriate, as it now holds columns for ROIs and events, not just `x`, `y`, and `Likelihood`. Maybe rename `coords` to `variables` (though that is quite vague).

Related to the above, the `scorer` value given by `deeplabcut` is what you have named in the `input_config.yaml` file as `pose_estimation_model_str`. We could use the same name in both places, or we could completely drop this column level, since with the current configuration it's only going to be one common value for everything.

In general, I think we should think how to best organize the output dataframe, without necessarily adhering to `deeplabcut` design choices. But it's hard for me to imagine what the best organization is for now. Will have to wait till we start actually using the df to make plots

Reply:
making a note of all this. I will start a new PR for re-structuring the columns of the exported dataframe (in the spirit of small and easily reviewable PRs)